Content.Client/Decals/DecalSystem.cs

```diff
@@ -0,0 +1,102 @@
+using System.Collections.Generic;
+using Content.Shared.Decals;
+using Robust.Client.Graphics;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+
+namespace Content.Client.Decals
+{
+    public class DecalSystem : SharedDecalSystem
+    {
+        [Dependency] private readonly IOverlayManager _overlayManager = default!;
+
+        private DecalOverlay _overlay = default!;
+        public Dictionary<GridId, SortedDictionary<int, SortedDictionary<uint, Decal>>> DecalRenderIndex = new();
+        private Dictionary<GridId, Dictionary<uint, int>> DecalZIndexIndex = new();
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            _overlay = new DecalOverlay(this, MapManager, PrototypeManager);
+            _overlayManager.AddOverlay(_overlay);
```

Comment:
Future concern but would be nice to have a console command to turn off decal rendering (for mapping purposes or otherwise).